plaidml/op/op_test.cc

```diff
@@ -176,6 +176,71 @@ TEST_F(OpTest, Concatenate) {
   runProgram(program);
 }
 
+TEST_F(OpTest, 3dConvolution) {
+  auto I = Placeholder(DType::FLOAT32, {1, 32, 224, 224, 3}, "I");
+  auto K = Placeholder(DType::FLOAT32, {7, 7, 7, 3, 64}, "K");
+  edsl::TensorRef O = edsl::Tensor(op::convolution(I, K).strides({2, 2, 2}).autopad_mode(op::AutoPadMode::EXPLICIT).manual_padding({2, 3, 2, 3, 2, 3}));
+  auto program = makeProgram("convolution", {O});
+  IVLOG(1, program);
+  std::vector<float> I_input(1*32*224*224*3, 0.0);
+  std::vector<float> K_filter(7*7*7*3*64, 0.0);
+  std::vector<float> O_output(1*16*112*112*64, 0.0);
+  edsl::TensorBuffers input_buffers{{I, I_input}, {K, K_filter}};
+  edsl::TensorBuffers output_buffers{{O, O_output}};
+  checkProgram(program, input_buffers, output_buffers);
```

Comment:
Only use `checkProgram` if you have actual inputs/outputs to validate against. `runProgram` is sufficient when you don't want to check correctness, it will supply the properly sized input/output buffers for you.

Reply:
Actually, I think it's because the workaround is to avoid using XSMM, and in this case, due to the strides, this one will run very slowly. So I don't think this is a compiler perf issue.